src/main/java/com/commercetools/sync/customers/utils/CustomerReferenceResolutionUtils.java

```diff
@@ -0,0 +1,180 @@
+     *        <td>{@link ResourceIdentifier}&lt;{@link Type}&gt;</td>
+     *     </tr>
+     *   </tbody>
+     * </table>
+     *
+     * <p><b>Note:</b> The {@link CustomerGroup} and {@link Type} references should be expanded with a key.
+     * Any reference that is not expanded will have its id in place and not replaced by the key will be
+     * considered as existing resources on the target commercetools project and
+     * the library will issues an update/create API request without reference resolution.
+     *
+     * @param customers the customers with expanded references.
+     * @return a {@link List} of {@link CustomerDraft} built from the supplied {@link List} of {@link Customer}.
+     */
+    @Nonnull
+    public static List<CustomerDraft> mapToCustomerDrafts(
+        @Nonnull final List<Customer> customers) {
+        return customers
+            .stream()
+            .map(CustomerReferenceResolutionUtils::mapToCustomerDraft)
+            .collect(toList());
+    }
+
+    @Nonnull
+    private static CustomerDraft mapToCustomerDraft(@Nonnull final Customer customer) {
+        return CustomerDraftBuilder
+            .of(customer.getEmail(), customer.getPassword())
+            .customerNumber(customer.getCustomerNumber())
+            .key(customer.getKey())
+            .firstName(customer.getFirstName())
+            .lastName(customer.getLastName())
+            .middleName(customer.getMiddleName())
+            .title(customer.getTitle())
+            .externalId(customer.getExternalId())
+            .companyName(customer.getCompanyName())
+            .customerGroup(getResourceIdentifierWithKey(customer.getCustomerGroup()))
+            .dateOfBirth(customer.getDateOfBirth())
+            .isEmailVerified(customer.isEmailVerified())
+            .vatId(customer.getVatId())
+            .addresses(customer.getAddresses())
+            .defaultBillingAddress(getAddressIndex(customer.getAddresses(), customer.getDefaultBillingAddressId()))
+            .billingAddresses(getAddressIndexList(customer.getAddresses(), customer.getBillingAddressIds()))
+            .defaultShippingAddress(getAddressIndex(customer.getAddresses(), customer.getDefaultShippingAddressId()))
+            .shippingAddresses(getAddressIndexList(customer.getAddresses(), customer.getShippingAddressIds()))
+            .custom(mapToCustomFieldsDraft(customer))
+            .locale(customer.getLocale())
+            .salutation(customer.getSalutation())
+            .stores(mapToStores(customer))
+            .build();
+    }
+
+
+    @Nullable
+    private static Integer getAddressIndex(
+        @Nullable final List<Address> allAddresses,
+        @Nullable final String addressId) {
+
+        if (allAddresses == null) {
+            return null;
+        }
+        if (isBlank(addressId)) {
+            return null;
+        }
+        for (int i = 0; i < allAddresses.size(); i++) {
+            String id = allAddresses.get(i).getId();
```

Comment:
Address element may have potential to be null. I know we did the check in validator. As a utility class we cannot assume it has been validated before it is manipulated. wdyt?